web/Services/LandlordService.cs

```diff
@@ -77,4 +78,19 @@ await using (var transaction = await _dbContext.Database.BeginTransactionAsync(I
 
         return ILandlordService.LandlordRegistrationResult.Success;
     }
+
+    public async Task ApproveLandlord(int landlordId,  BricksAndHeartsUser user)
+    {
+        var landlord = _dbContext.Landlords.Single(l =>l.Id == landlordId);
+        landlord.CharterApproved = true;

```

Comment:
Should we check whether they have been approved already?

I think we probably should. Just throwing an exception in that case is fine I think, though if you prefer you could handle it more nicely with a redirect and flash message.

Reply:
The method should only be callable on landlords who have not been approved (the 'Approve' button does not appear on the profile of those who aren't unapproved), but I can add a second check here if you think it's important.

Reply:
I think it's worth having, e.g. someone could have two tabs open, and do it in one and then try and do it again in another, which would then overwrite the approval time. Not the end of the world, but not really desired behaviour.